piano-analytics/src/main/java/io/piano/android/analytics/MediaHelper.kt

```diff
@@ -113,6 +115,16 @@ class MediaHelper internal constructor(
         bufferHeartbeatDurations.replaceValues(values, MIN_BUFFER_HEARTBEAT_DURATION)
     }
 
+    /**
+     * Sets a map of extraProps on the media object and returns it.
+     * @param extraProps The extra props (e.g. av_content properties)
+     * @return this [MediaHelper] instance
+     */
+    @Suppress("unused", "MemberVisibilityCanBePrivate") // Public API.
+    fun setExtraProps(extraProps: Array<out Property>) = apply {
+        this.extraProps = extraProps
```

Comment:
Array values can be modified outside, it's better to make a copy here via `extraProps.copyOf()`